### ISSUE TITLE
implementation details about the testing process

### ISSUE THREAD

**Issue by jiauxan** [GANPerf/LCR#5]
I ran the code of LCR you provided, pretraining the model on the CUB dataset for 100 epochs and then freezing the model parameters and only optimizing the linear classiﬁer . However, I could only achieve a maximum acc1 around 64.71. I would like to know if there is any step I might have missed or done incorrectly. thanks a lot!

**Comment by eafn** [comment]
me too

**Comment by andayangyang** [comment]
> I ran the code of LCR you provided, pretraining the model on the CUB dataset for 100 epochs and then freezing the model parameters and only optimizing the linear classiﬁer . However, I could only achieve a maximum acc1 around 64.71. I would like to know if there is any step I might have missed or done incorrectly. thanks a lot!

I appreciate your interest. Could you please let me know which checkpoint you have been using? How about the acc1 performance on StanfordCars & Aircraft? To attain the highest acc1 accuracy, it is advisable to consider selecting the checkpoint with the best retrieval performance, rather than relying solely on the last epoch during the pretraining process. For your reference, I have provided the checkpoint at the following link: [Checkpoint Link](https://drive.google.com/uc?export=download&id=15zimj88rQCWWa7y5K2p9rUX7QrUh6Dpf)

**Comment by GANPerf** [comment]
> I ran the code of LCR you provided, pretraining the model on the CUB dataset for 100 epochs and then freezing the model parameters and only optimizing the linear classiﬁer . However, I could only achieve a maximum acc1 around 64.71. I would like to know if there is any step I might have missed or done incorrectly. thanks a lot!

I appreciate your interest. Could you please let me know which checkpoint you have been using? How about the acc1 performance on StanfordCars & Aircraft?  To attain the highest acc1 accuracy, it is advisable to consider selecting the checkpoint with the best retrieval performance, rather than relying solely on the last epoch during the pretraining process. For your reference, I have provided the checkpoint at the following link: [Checkpoint Link](https://drive.google.com/uc?export=download&id=15zimj88rQCWWa7y5K2p9rUX7QrUh6Dpf)

**Comment by jiauxan** [comment]
> 

Thank you for your reply! I only test on the CUB dataset and select the checkpoint with the best retrieval performance of the result of main_moco.py(the best retrieval performance gained around 35 epochs and reached acc around 42). Using the checkpoint on the last epoch to gain a linear classifier always attains the result around 62. The parameters I set are the same as you have given in the paper.

**Comment by jiauxan** [comment]
Another question I wonder regarding training is about loading the pre-trained model parameters you have provided. I modified the pre-trained model instead of the one you have given. However, the size of the fc is mismatched with the LCR. The num-classes for resnet-50 is 1000 instead of 256 you have given. I would like to know whether the parameters of the checkpoint you have provided should be loaded by resnet-50(encoder_q) instead of the LCR， and the error happened during I used moco_linear.py.
![image](https://github.com/GANPerf/LCR/assets/43083273/f383c924-490c-4143-b1ac-16cd99da9d46)

**Comment by GANPerf** [comment]
> Another question I wonder regarding training is about loading the pre-trained model parameters you have provided. I modified the pre-trained model instead of the one you have given. However, the size of the fc is mismatched with the LCR. The num-classes for resnet-50 is 1000 instead of 256 you have given. I would like to know whether the parameters of the checkpoint you have provided should be loaded by resnet-50(encoder_q) instead of the LCR， and the error happened during I used moco_linear.py. ![image](https://user-images.githubusercontent.com/43083273/248834724-f383c924-490c-4143-b1ac-16cd99da9d46.png)

Hi Jiaxuan, I wanted to mention that I believe the LCR (2048->256) modification would be more suitable because the ResNet-50 model has a different structure (2048->1000).

**Comment by jiauxan** [comment]
> > Another question I wonder regarding training is about loading the pre-trained model parameters you have provided. I modified the pre-trained model instead of the one you have given. However, the size of the fc is mismatched with the LCR. The num-classes for resnet-50 is 1000 instead of 256 you have given. I would like to know whether the parameters of the checkpoint you have provided should be loaded by resnet-50(encoder_q) instead of the LCR， and the error happened during I used moco_linear.py. ![image](https://user-images.githubusercontent.com/43083273/248834724-f383c924-490c-4143-b1ac-16cd99da9d46.png)
> 
> Hi Jiaxuan, I wanted to mention that I believe the LCR (2048->256) modification would be more suitable because the ResNet-50 model has a different structure (2048->1000).


Hi, I also agree with that. But it seems that the size of fc in the model in the checkpoint you have provided mismatched the model in your code(I mean the LCR model), as if in the pic above. I wonder whether the checkpoint you have provided is gained using main_moco.py.

**Comment by shisofia** [comment]
Hello, I have encountered the same problem. May I ask if you have solved this problem and could you tell me your solution? @jiauxan
